src/common/models/Projection.ts

```diff
@@ -521,17 +517,16 @@ export class Projection {
       ] as MetricsTimeseriesRow;
       actualsTimeseries.push(actualsTimeseriesrowForDate || null);
       metricsTimeseries.push(metricsTimeseriesRowForDate || null);
-      dates.push(currDate.toDate());
+      // Clone the date since we're about to mutate it below.
+      dates.push(new Date(currDate.getTime()));
 
       // increment the date by one
-      currDate = currDate.clone().add(1, 'days');
+      currDate.setUTCDate(currDate.getUTCDate() + 1);
     }
 
-    // only keep futureDaysToInclude days ahead of today
     const now = new Date();
     const todayIndex = dates.findIndex(date => date > now);
-    const days =
-      todayIndex >= 0 ? todayIndex + futureDaysToInclude : dates.length;
+    const days = todayIndex >= 0 ? todayIndex : dates.length;
```

Comment:
Not very sure whether we still need `days` variable as is, since `futureDaysToInclude` has been removed?

Reply:
I think you can just remove `now`, `todayIndex`, `days` and the calls to `slice()`.